skopt/utils.py

```diff
@@ -292,3 +295,50 @@ def cook_estimator(base_estimator, space=None, **kwargs):
 
     base_estimator.set_params(**kwargs)
     return base_estimator
+
+def transform_dims(dimensions, estimator="GP"):
+    """
+    Transforms space for Gaussian processes.
+
+    Parameters
+    ----------
+    * `dimensions` [list, shape=(n_dims,)]:
+        List of search space dimensions.
+        Each search dimension can be defined either as
+
+        - a `(upper_bound, lower_bound)` tuple (for `Real` or `Integer`
+          dimensions),
+        - a `(upper_bound, lower_bound, "prior")` tuple (for `Real`
+          dimensions),
+        - as a list of categories (for `Categorical` dimensions), or
+        - an instance of a `Dimension` object (`Real`, `Integer` or
+          `Categorical`).
+
+         NOTE: The upper and lower bounds are inclusive for `Integer`
+         dimensions.
+    * `estimator` [str, ]:
+        optimization estimator to be used.
+        each estimator may use different transformations for dimensions.
+    """
```

Comment:
Maybe we can add an example here to show how to use it together with `Optimizer`. Would that be useful for people who don't know the insides of `gp_minimize`?